src/software/thunderscope/binary_context_managers/simulator.py

```diff
@@ -0,0 +1,364 @@
+        :param value: The exception that was raised
+        :param traceback: The traceback of the exception
+
+        """
+        if self.er_force_simulator_proc:
+            self.er_force_simulator_proc.kill()
+            self.er_force_simulator_proc.wait()
+
+    def setup_proto_unix_io(
+        self,
+        simulator_proto_unix_io,
+        blue_full_system_proto_unix_io,
+        yellow_full_system_proto_unix_io,
+    ):
+
+        """Setup the proto unix io for the simulator
+
+        :param simulator_proto_unix_io: The proto unix io of the simulator.
+        :param blue_full_system_proto_unix_io: The proto unix io of the blue full system.
+        :param yellow_full_system_proto_unix_io: The proto unix io of the yellow full system.
+
+        """
+
+        # inputs to er_force_simulator_main
+        for arg in [
+            (SIMULATION_TICK_PATH, SimulatorTick),
+            (WORLD_STATE_PATH, WorldState),
+        ]:
+            simulator_proto_unix_io.attach_unix_sender(self.simulator_runtime_dir, *arg)
+
+        simulator_proto_unix_io.attach_unix_receiver(
+            self.simulator_runtime_dir,
+            WORLD_STATE_RECEIVED_TRIGGER_PATH,
+            WorldStateReceivedTrigger,
+        )
+
+        # setup blue full system unix io
+        for arg in [
+            (BLUE_WORLD_PATH, World),
+            (BLUE_PRIMITIVE_SET, PrimitiveSet),
+        ]:
+            blue_full_system_proto_unix_io.attach_unix_sender(
+                self.simulator_runtime_dir, *arg
+            )
+
+        for arg in [
+            (BLUE_SSL_WRAPPER_PATH, SSL_WrapperPacket),
+            (BLUE_ROBOT_STATUS_PATH, RobotStatus),
+            (SIMULATOR_STATE_PATH, SimulatorState),
+        ] + [
+            # TODO (#2655): Add/Remove HRVO layers dynamically based on the HRVOVisualization proto messages
+            (BLUE_HRVO_PATH, HRVOVisualization, True)
+            for _ in range(MAX_ROBOT_IDS_PER_SIDE)
+        ]:
+            blue_full_system_proto_unix_io.attach_unix_receiver(
+                self.simulator_runtime_dir, *arg
+            )
+
+        # setup yellow full system unix io
+        for arg in [
+            (YELLOW_WORLD_PATH, World),
+            (YELLOW_PRIMITIVE_SET, PrimitiveSet),
+        ]:
+            yellow_full_system_proto_unix_io.attach_unix_sender(
+                self.simulator_runtime_dir, *arg
+            )
+
+        for arg in [
+            (YELLOW_SSL_WRAPPER_PATH, SSL_WrapperPacket),
+            (YELLOW_ROBOT_STATUS_PATH, RobotStatus),
+        ] + [
+            # TODO (#2655): Add/Remove HRVO layers dynamically based on the HRVOVisualization proto messages
+            (YELLOW_HRVO_PATH, HRVOVisualization, True)
+            for _ in range(MAX_ROBOT_IDS_PER_SIDE)
+        ]:
+            yellow_full_system_proto_unix_io.attach_unix_receiver(
+                self.simulator_runtime_dir, *arg
+            )
+
+
+class Gamecontroller(object):
```

Comment:
This seems to be duplicated and removed from this file

Reply:
fixed.